tests/test_types_basic.py

```diff
@@ -74,12 +74,15 @@ def testDecimal(self):
         self.failUnless(str(s) == "NaN", "wrong decimal quoting: " + str(s))
         self.failUnless(type(s) == decimal.Decimal,
                         "wrong decimal conversion: " + repr(s))
+
+    @testutils.skip_before_postgres(14, 0)
```

Comment:
`SELECT 'Infinity'::numeric AS foo` causes

```
psycopg2.errors.InvalidTextRepresentation: invalid input syntax for type numeric: "Infinity"
```

in Postgres 12 and 13.

Reply:
Oh, now that I think about it more we should probably rather check the server version when adapting `Decimal('Infinity')`, and fall back to `NaN'::numeric` on Postgres < 14.

Reply:
Uhm... Thinking a bit harder abut it: is it even a good idea to change the way inf is converted? Yes, converting it to nan wasn't the right thing to do, but maybe there is code that assumes that behaviour.

I am reconsidering the opportunity of making this change.

Note: psycopg 3 doesn't check for version and doesn't convert inf -> nan. Trying to send an inf results in a server error, which is the right thing to do.

Reply:
I agree. Either option (converting it or failing) is a potential breaking change. Happy to close this if you think we err on preserving current behavior.